### PR TITLE
AT-7049: New OpenAPI specification for provisioning

### PR DESCRIPTION
**Pull request by jeffsegal-ccpo**
Defines a new OpenAPI specification which will be used to allow external machine interfaces (initially just SNOW) to invoke provisioning operations in our AWS infrastructure. This will support the following use cases:

- Adding new ~~accounts~~ portfolios
- Adding funding sources to existing ~~accounts~~ portfolios
- Adding operators to existing ~~accounts~~ portfolios

It will be tied very closely to the public ATAT spec that the CSPs will implement and share some data structures (in the future we could look at cleaner ways to do so), but it won't necessarily be a 1:1 copy. It is fairly simple for now, but if we add more capability to the public spec, that will need to be reflected here as well.

### CODE REVIEW

File: atat_provisioning_api.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      description: Payload for the addition of an Operator to an existing CSP Account
+      allOf:
+        - $ref: '#/components/schemas/ProvisioningPayload'
+      properties:
+        operators:
+          type: array
+          description: Array of operator email addresses
+          items:
+            type: string
+            example: "user@mail.mil"
```
[the remark this answers is not in the document]
Comment (jeffsegal-ccpo) [reply atat_provisioning_api.yaml:173]: Not at all silly. That was part of the effort to simplify the provisioning process. My proposal is that we will simply collect a set of email addresses who will receive full access to the Accounts and call it a day.

File: atat_provisioning_api.yaml [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+          $ref: '#/components/schemas/CspResponse'
+          readOnly: true
+    OperationType:
+      description: Distinguishes type of operation that a user is performing
+      type: string
+      enum:
+        - ADD_PORTFOLIO
+        - ADD_OPERATORS
+        - ADD_FUNDING_SOURCE
+    TargetCloud:
```
Comment (jasonburkert-ccpo) [inline atat_provisioning_api.yaml:145]: As we discussed via call.  Unsure about meaning of _cloud_ vs. _csp_ terminology.

```suggestion
    TargetCloudServiceProvider:
```

**Review by jasonburkert-ccpo — CHANGES_REQUESTED** [review]
Couple minor suggestions and one about cloud/csp terminology that we spoke about.

File: atat_provisioning_api.yaml [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+              schema:
+                $ref: '#/components/schemas/ProvisioningRequest'
+        '400':
+          description: Invalid input
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ValidationError'
+      requestBody:
+        description: An empty String or JSON body
```
[the remark this answers is not in the document]
Comment (jeffsegal-ccpo) [reply atat_provisioning_api.yaml:49]: Fixed

File: atat_provisioning_api.yaml [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+      content: {}
+  schemas:
+    Error:
+      type: object
+      additionalProperties: false
+      description: Generic error model
+      properties:
+        code:
+          type: string
+          enum: [INVALID_INPUT, OTHER]
```
[the remark this answers is not in the document]
Comment (jeffsegal-ccpo) [reply atat_provisioning_api.yaml:84]: Yeah if we're not using it for any sort of automation purposes then it's probably better to leave it out.

**Review by jasonburkert-ccpo — APPROVED** [review]

**Pull request merged**